nifi-nar-bundles/nifi-elasticsearch-bundle/nifi-elasticsearch-client-service/src/test/java/org/apache/nifi/elasticsearch/integration/ElasticSearchClientService_IT.java

```diff
@@ -0,0 +1,741 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assumptions.assumeTrue;
+
+public class ElasticSearchClientService_IT extends AbstractElasticsearch_IT {
+    private TestRunner runner;
+    private ElasticSearchClientServiceImpl service;
+
+    static final String INDEX = "messages";
+    public static String TYPE;
+
+    private static final ObjectMapper MAPPER = new ObjectMapper();
+
+    private static TlsConfiguration generatedTlsConfiguration;
+    private static TlsConfiguration truststoreTlsConfiguration;
+
+    @BeforeAll
+    static void beforeAll() throws IOException {
+        startTestcontainer();
+        TYPE = getElasticMajorVersion() == 6 ? "_doc" : "";
+        System.out.println(
+                String.format("%n%n%n%n%n%n%n%n%n%n%n%n%n%n%nTYPE: %s%n%s%s%n%n%n%n%n%n%n%n%n%n%n%n%n%n%n",
+                        TYPE, IMAGE.getRepository(), IMAGE.getVersionPart())
+        );
+
+        generatedTlsConfiguration = new TemporaryKeyStoreBuilder().build();
+        truststoreTlsConfiguration = new StandardTlsConfiguration(
+                null,
+                null,
+                null,
+                generatedTlsConfiguration.getTruststorePath(),
+                generatedTlsConfiguration.getTruststorePassword(),
+                generatedTlsConfiguration.getTruststoreType()
+        );
+
+        setupTestData();
+    }
+
+    @AfterAll
+    public static void afterAll() throws IOException {
+        tearDownTestData();
+        stopTestcontainer();
+    }
+
+    @BeforeEach
+    void before() throws Exception {
+        runner = TestRunners.newTestRunner(TestControllerServiceProcessor.class);
+        service = new ElasticSearchClientServiceImpl();
+        runner.addControllerService("Client Service", service);
+        runner.setProperty(service, ElasticSearchClientService.HTTP_HOSTS, ELASTIC_HOST);
+        runner.setProperty(service, ElasticSearchClientService.CONNECT_TIMEOUT, "10000");
+        runner.setProperty(service, ElasticSearchClientService.SOCKET_TIMEOUT, "60000");
+        runner.setProperty(service, ElasticSearchClientService.RETRY_TIMEOUT, "60000");
+        runner.setProperty(service, ElasticSearchClientService.SUPPRESS_NULLS, ElasticSearchClientService.ALWAYS_SUPPRESS.getValue());
+        runner.setProperty(service, ElasticSearchClientService.USERNAME, "elastic");
+        runner.setProperty(service, ElasticSearchClientService.PASSWORD, ELASTIC_USER_PASSWORD);
+
+        try {
+            runner.enableControllerService(service);
+        } catch (Exception ex) {
+            throw ex;
+        }
+
+        service.refresh(null, null);
+    }
+
+    @AfterEach
+    void after() throws Exception {
+        service.onDisabled();
+    }
+
+    private String prettyJson(Object o) throws JsonProcessingException {
+        return MAPPER.writerWithDefaultPrettyPrinter().writeValueAsString(o);
+    }
+
+    private class MapBuilder {
```

Comment:
Like this as a solution while we're stuck with Java 8, but I wonder whether it's better to have an `of` method instead of `put`, which would make it simpler to migrate to the Java 11 `Map.of` API in future?

Could also change `build` to wrap the Map in a `Collections.unmodifiableMap` to again be closer to `Map.of` implementation and ensure the constructed Map isn't changed unintentionally by the tests?

Is it also worth making this a higher-level class somewhere in a test NAR so that others could use it (particularly when replaceing Groovy tests)?

Happy for you to ignore any/all of these suggestions

Reply:
Made that change.